src/main/java/org/opensearch/security/resources/sharing/ResourceSharing.java

```diff
@@ -244,6 +254,37 @@ public boolean isSharedWithEntity(Recipient recipientType, Set<String> targets,
         return shareWith.atAccessLevel(accessLevel).isSharedWithAny(recipientType, targets);
     }
 
+    /**
+     * Resolves all access levels that the given {@link User} is entitled to.
+     * <p>
+     * This method aggregates access levels based on:
+     * <ul>
+     *   <li>The user’s explicit identifier (username and wildcard {@code *}).</li>
+     *   <li>The user’s security roles (including the wildcard {@code *}).</li>
+     *   <li>The user’s backend roles (including the wildcard {@code *}).</li>
+     * </ul>
+     * For each category (user, roles, backend roles), a lookup is performed through
+     * {@link #fetchAccessLevels(Recipient, Set)} to collect the matching access levels.
+     * </p>
+     *
+     * @param user the {@link User} whose access levels should be determined;
+     *             must not be {@code null}.
+     * @return a {@link Set} of access level identifiers granted to the user, never {@code null}.
+     */
+    public Set<String> getAccessLevelsForUser(User user) {
+        Set<String> userRoles = new HashSet<>(user.getSecurityRoles());
+        Set<String> userBackendRoles = new HashSet<>(user.getRoles());
+
+        userRoles.add("*");
+        userBackendRoles.add("*");
```

Comment:
i understand that this was copied from ResouceAccessHandler but we probly need to revisit the wildcards here as they may not be needed. 
we also need to refactor fetchAccessLevels if we plan to go with user:* as the only convention for public resources.

Reply:
yes, I'd like to revisit this as well. I do think we want to have the ability to make public at different levels. i.e. public read only vs public anyone can modify.